### PR TITLE
feat(bigquery): add json type support

### PR DESCRIPTION
**Pull request by NivedhaSenthil**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/google-cloud-ruby/issues) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea.
- [x] Follow the instructions in [CONTRIBUTING](CONTRIBUTING.md). Most importantly, ensure the tests and linter pass by running `bundle exec rake ci` in the gem subdirectory.
- [x] Update code documentation if necessary.

closes: #20518

### CODE REVIEW

File: google-cloud-bigquery/lib/google/cloud/bigquery/schema/field.rb [hunk 1]
```diff
@@ -456,6 +459,15 @@ def geography?
             type == "GEOGRAPHY"
           end
 
+          ##
+          # Checks if the type of the field is `JSON`.
+          #
+          # @return [Boolean] `true` when `GEOGRAPHY`, `false` otherwise.
```
Comment (shivgautam) [inline google-cloud-bigquery/lib/google/cloud/bigquery/schema/field.rb:465]: Should be 'when `JSON`' instead

File: google-cloud-bigquery/acceptance/bigquery/named_params_test.rb [hunk 2]
```diff
@@ -232,6 +232,31 @@
...
+
+    _(rows.class).must_equal Google::Cloud::Bigquery::Data
+    _(rows.fields.count).must_equal 1
+    _(rows.fields.first.name).must_equal "value"
+    _(rows.fields.first).must_be :json?
+    _(rows.count).must_equal 1
+    _(JSON.parse(rows.first[:value])).must_equal value
+  end
+
+  it "queries the data with a nil parameter and geography type" do
```
[the remark this answers is not in the document]
Comment (NivedhaSenthil) [reply google-cloud-bigquery/acceptance/bigquery/named_params_test.rb]: Nope, test template copied which was not updated :| Updated it now.

**Review by shivgautam — APPROVED** [review]
LGTM. Thank you!

**Comment by shivgautam** [comment]
@NivedhaSenthil - Do we need to add samples for JSON usage or is the code documentation sufficient?

**Comment by NivedhaSenthil** [comment]
There is no region tag I could find and the doc doesn't seem to have samples too https://cloud.google.com/bigquery/docs/json-data

File: google-cloud-bigquery/lib/google/cloud/bigquery/convert.rb [hunk 3]
```diff
@@ -72,7 +73,7 @@ def self.format_value value, field
             value[:v].map { |v| format_value v, field }
           elsif Hash === value[:v]
             format_row value[:v], field.fields
-          elsif field.type == "STRING"
+          elsif field.type == "STRING" || field.type == "JSON" || field.type == "GEOGRAPHY"
```
Comment (ajupazhamayil) [inline google-cloud-bigquery/lib/google/cloud/bigquery/convert.rb:76]: GEOGRAPHY is newly added?

**Pull request merged**